src/polygon/src/polygon.cpp

```diff
@@ -155,6 +158,46 @@ void CheckGeometry(const std::vector<glm::ivec3> &triangles,
                                   vertPos[tri[2]], precision) >= 0;
                      }),
          geometryErr, "triangulation is not entirely CCW!");
+
+  std::vector<std::pair<glm::vec2, glm::vec2>> edges(3 * triangles.size());
+  std::for_each(
+#if MANIFOLD_PAR == 'T' && !(__APPLE__)
+      std::execution::par,
+#endif
+      countAt(0ul), countAt(triangles.size()), [&](size_t i) {
+        edges[3 * i] =
+            std::make_pair(vertPos[triangles[i].x], vertPos[triangles[i].y]);
+        edges[3 * i + 1] =
+            std::make_pair(vertPos[triangles[i].y], vertPos[triangles[i].z]);
+        edges[3 * i + 2] =
+            std::make_pair(vertPos[triangles[i].z], vertPos[triangles[i].x]);
+      });
+  std::mutex mutex;
+  std::vector<std::pair<size_t, size_t>> overlappingPairs;
+  auto p = [&](size_t i) {
+    return std::all_of(countAt(i + 1), countAt(edges.size()), [&](size_t j) {
+      if (intersect(edges[i].first, edges[i].second, edges[j].first,
```

Comment:
Related to my comment on your last PR - I don't *think* you need to check intersection of edges. I believe a manifold triangulation has every triangle CCW if and only if no triangles overlap/intersect. That's why I never had this intersection check before. Have you found a counter example?

Reply:
We did have a counter example previously: https://github.com/elalish/manifold/issues/502#issuecomment-1653968014

Not sure if this can still happen after #508, but I think it doesn't hurt to check? The check is not too slow (comparing with the original `checkGeometry`.)

Reply:
An interesting thing I found is that, some triangulation is only epsilon-valid but not strictly valid. I guess this is a consequence of the input mesh/polygon being only epsilon-valid?

Reply:
Indeed, one of them is CW. I guess I can remove this check.